controllers/priorityexpander_controller.go

```diff
@@ -0,0 +1,240 @@
+	for _, node := range status.NodeGroups {
+		oroot[node.Name] = make(map[string]int32)
+		oroot[node.Name]["CloudProviderTarget"] = node.Health.CloudProviderTarget
+		oroot[node.Name]["Ready"] = node.Health.Ready
+		oroot[node.Name]["Unready"] = node.Health.Unready
+		oroot[node.Name]["NotStarted"] = node.Health.NotStarted
+		oroot[node.Name]["LongNotStarted"] = node.Health.LongNotStarted
+		oroot[node.Name]["Registered"] = node.Health.Registered
+		oroot[node.Name]["LongUnregistered"] = node.Health.LongUnregistered
+		oroot[node.Name]["MinSize"] = node.Health.MinSize
+		oroot[node.Name]["MaxSize"] = node.Health.MaxSize
+	}
+
+	//Create new PriorityExpander template and parse it
+	t, err := template.New("template").Parse(pexp.Spec.Template)
+	if err != nil {
+		log.Error(
+			err,
+			"Error parsing PriorityExpander template. Check your syntax and/or rtfm.",
+		)
+		return ctrl.Result{}, err
+	}
+
+	buf := new(bytes.Buffer)
+	if err := t.Execute(buf, oroot); err != nil {
+		log.Error(
+			err,
+			"Error parsing generating template output.",
+		)
+		return ctrl.Result{}, err
+	}
+
+	// parsed content: fmt.Println(buf.String())
+	// Create the new ConfigMap object
+	pecm := kcore_v1.ConfigMap{
+		TypeMeta: kmeta_v1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: kmeta_v1.ObjectMeta{
+			Name:      r.Configuration.ClusterAutoscalerPEConfigMapName,
+			Namespace: r.Configuration.ClusterAutoscalerStatusNamespace,
+		},
+		Data: map[string]string{
+			"priorities": buf.String(),
+		},
+	}
+
+	// Update it inplace. The ConfigMap _must_ exist.
+	if err := r.Update(ctx, &pecm); err != nil {
+		log.Error(
+			err,
+			"Unable to update ClusterAutoscaler priority expander configmap",
+			"namespace", r.Configuration.ClusterAutoscalerStatusNamespace,
+			"name", r.Configuration.ClusterAutoscalerStatusName,
+		)
+
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PriorityExpanderReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1alpha1.PriorityExpander{}).
+		Watches(
+			&source.Kind{Type: &kcore_v1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(func(_ client.Object) []reconcile.Request {
+				ctx := context.Background()
+				log := ctrllog.Log.WithName("priorityexpander")
+
+				// List PriorityExpanders, even though there should be only one.
+				pexp := &corev1alpha1.PriorityExpanderList{}
+				if err := r.List(ctx, pexp); err != nil {
+					log.Error(err, "Unable to list PriorityExpanders")
+					return []reconcile.Request{}
+				}
+
+				// We reconcile all Priority Expander.
+				res := make([]reconcile.Request, len(pexp.Items))
+				for i, e := range pexp.Items {
+					res[i] = reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Namespace: e.Namespace,
+							Name:      e.Name,
+						},
+					}
+				}
```

Comment:
Why not just reconcile the `PriorityExpander` matching ClusterAutoscalerPEConfigMapName ?